src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,74 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public class Node<K , V>{
+        private int hash;
+        private K key;
+        private V value;
+        Node<K, V> next;
+
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+    private  Node<K, V> tail;
+
+    public static final int DEFAULT_CAPACITY = 16;
+
+    private Node<K, V>[] elements = new Node[DEFAULT_CAPACITY];
+
+    private int count;
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = new Node<>(key,value);
+        if (count == 0){
+            elements[0] = newNode;
+            tail = newNode;
+        } else if (count > DEFAULT_CAPACITY){
+            elements = resize(elements);
+
+        } else {
+            int index = getIndex(key);
+            elements[index] = newNode;
+            tail.next = newNode;
```

Comment:
Setting `tail.next = newNode` is incorrect if the new node is not being added at the end of the list. This logic should be revised to correctly maintain the linked list structure.